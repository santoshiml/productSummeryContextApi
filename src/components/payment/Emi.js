
import React from "react";
import { Form, Button } from "react-bootstrap";

function Emi(props) {
  const [disabled, enabled] = React.useState("disabled");
  
  const Continue = () => {
    props.switchpayment(2);
  };

  function Insert(val) {
    if (val != null) {
      enabled();
    }
    props.item(val);
  }

  return (
    <div>
      <h5 className="termsHeading">Choose your terms</h5>
      <Form.Check type="radio" className="emi-selector">
        <Form.Check.Label className="formEmiLable">
          <Form.Check.Input name="emi" type="radio" onClick={() => Insert(4327102)} />
          <div className="formEmiLeft">1 Payment due</div>
          <div className="formEmiRight">€43,271.02</div>
        </Form.Check.Label>
      </Form.Check>
      <Form.Check type="radio" className="emi-selector" >
        <Form.Check.Label className="formEmiLable">
          <Form.Check.Input name="emi" type="radio" onClick={() => Insert(2163551)} />
          <div className="formEmiLeft">2 Payment due</div>
          <div className="formEmiRight">€21,635.51</div>
        </Form.Check.Label>
      </Form.Check>
      <Form.Check type="radio" className="emi-selector" >
        <Form.Check.Label className="formEmiLable">
          <Form.Check.Input name="emi" type="radio" onClick={() => Insert(1442367)} />
          <div className="formEmiLeft">3 Payment due</div>
          <div className="formEmiRight">€14,423,67</div>
        </Form.Check.Label>
      </Form.Check>
      <Button name="submit" className={`continueBtn ${disabled}`} onClick={Continue}>
        Continued
      </Button>
    </div>
  );
}

export default Emi;
