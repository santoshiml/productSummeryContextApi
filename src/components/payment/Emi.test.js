import {mount} from 'enzyme'
import renderer from 'react-test-renderer';
import Emi from './Emi'

describe('emi cOMPONENT', () => {
    
    const wrapper = mount(<Emi />);

    it('Should Render Properly',()=>{
        const tree=renderer.create(<Emi/>).toJSON()
        expect(tree).toMatchSnapshot();

    })

    //can check all Input fields

    it('should render an email input tag', () => {
        expect(wrapper.find('input[name="emi"]').exists()).toBe(true);
    });

    it('check for class..',()=>{
        expect(wrapper.exists('.emi-selector')).toEqual(true)
    })

})