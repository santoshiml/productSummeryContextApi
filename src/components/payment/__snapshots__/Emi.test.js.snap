// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`emi cOMPONENT Should Render Properly 1`] = `
<div>
  <h5
    className="termsHeading"
  >
    Choose your terms
  </h5>
  <div
    className="emi-selector form-check"
  >
    <label
      className="formEmiLable form-check-label"
    >
      <input
        className="form-check-input"
        name="emi"
        onClick={[Function]}
        type="radio"
      />
      <div
        className="formEmiLeft"
      >
        1 Payment due
      </div>
      <div
        className="formEmiRight"
      >
        €43,271.02
      </div>
    </label>
  </div>
  <div
    className="emi-selector form-check"
  >
    <label
      className="formEmiLable form-check-label"
    >
      <input
        className="form-check-input"
        name="emi"
        onClick={[Function]}
        type="radio"
      />
      <div
        className="formEmiLeft"
      >
        2 Payment due
      </div>
      <div
        className="formEmiRight"
      >
        €21,635.51
      </div>
    </label>
  </div>
  <div
    className="emi-selector form-check"
  >
    <label
      className="formEmiLable form-check-label"
    >
      <input
        className="form-check-input"
        name="emi"
        onClick={[Function]}
        type="radio"
      />
      <div
        className="formEmiLeft"
      >
        3 Payment due
      </div>
      <div
        className="formEmiRight"
      >
        €14,423,67
      </div>
    </label>
  </div>
  <button
    className="continueBtn disabled btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Continued
  </button>
</div>
`;
